bin/android-functions.sh

```diff
@@ -4,6 +4,261 @@ declare root="$(cd "$(dirname "$(dirname "${BASH_SOURCE[0]}")")" && pwd)"
+        fi
+      fi
+    done
+  done
+
+  temp=$(mktemp)
+  for java_home_test in "${JAVA_HOME_SEARCH_PATHS[@]}"; do
+    # Try initial search in bin of known location before attempting `find`
+    write_log "v" "Checking $java_home_test/bin/javac$exe"
+    if [[ -f "$java_home_test/bin/javac$exe" ]]; then
+
+      test_javac_version "$java_home_test/bin/javac$exe" "$JDK_VERSION" ; r=$?
+      if [[ "$r" == "0" ]]; then
+        # subshell, output to file
+        echo "$(dirname "$(dirname "$javac")")" > "$temp"
+        echo "$java_home_test" > "$temp"
+        break
+      else
+        write_log "v" "Ignoring a $java_home_test/bin/javac$exe $jc_v"
+      fi
+    fi
+
+    write_log "v" "find $java_home_test -type f -name "javac$exe"' -print0 2>/dev/null | while IFS= read -r -d '' javac"
+    find "$java_home_test" -type f -name "javac$exe" -print0 2>/dev/null | while IFS= read -r -d '' javac
+    do
+      # break doesn't work here, check that we don't have a result
+      if [[ $(stat_size "$temp") == 0 ]]; then
+        write_log "v" "Found $javac"
+        test_javac_version "$javac" "$JDK_VERSION" ; r=$?
+        if [[ "$r" == "0" ]]; then
+          # subshell, output to file
+          echo "$(dirname "$(dirname "$javac")")" > "$temp"
+        else
+          write_log "v" "Ignoring b $javac $jc_v"
+        fi
+      fi
+    done
+  done
+
+  if [[ $(stat_size "$temp") != 0 ]]; then
+    _jh=$(cat "$temp")
+  fi
+
+  echo -n > "$temp"
+
+  for gradle_test in "${GRADLE_SEARCH_PATHS[@]}"; do
+    write_log "v" "Checking $gradle_test/bin/gradle$bat"
+    if [[ -f "$gradle_test/bin/gradle$bat" ]]; then
+      echo "$gradle_test" > "$temp"
+      break
+    fi
+
+    write_log "v" "find $gradle_test -type f -name 'gradle' -print0 2>/dev/null | while IFS= read -r -d '' gradle"
+    find "$gradle_test" -type f -name "gradle$bat" -print0 2>/dev/null | while IFS= read -r -d '' gradle
+    do
+      # break doesn't work here, check that we don't have a result
+      if [[ $(stat_size "$temp") == 0 ]]; then
+        write_log "v" "Found $gradle"
+        # subshell, output to file
+        echo "$(dirname "$(dirname "$gradle")")" > "$temp"
+      fi
+    done
+  done
+
+  if [[ $(stat_size "$temp") != 0 ]]; then
+    _gh=$(cat "$temp")
+  fi
+  rm "$temp"
+
+  if [[ -n "$_ah" ]]; then
+    ANDROID_HOME="$(native_path "$_ah")"
+    export ANDROID_HOME
+  fi
+  
+  if [[ -n "$_sdk" ]]; then
+    ANDROID_SDK_MANAGER="$(native_path "$_sdk")"
+    export ANDROID_SDK_MANAGER
+  fi
+  
+  if [[ -n "$_jh" ]]; then
+    JAVA_HOME="$(native_path "$_jh")"
+    export JAVA_HOME
+  fi
+
+  
+  if [[ -n "$_gh" ]]; then
+    GRADLE_HOME="$(native_path "$_gh")"
+    export GRADLE_HOME
+  fi
+}
+
+function build_android_platform_tools_uri() {
+  local os="$(host_os)"
+
+  uri=${ANDROID_PLATFORM_TOOLS_URI_TEMPLATE/\{os\}/"$(android_host_platform "$os")"}
+  echo "$uri"
+}
+
+function build_android_command_line_tools_uri() {
+  local os="$(host_os)"
```

Comment:
```sh
  local os="$(lower "$(host_os)")"
```

Reply:
Have resolved this on android-fte branch as that is where the incorrect code was introduced.